File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, str: str) -> bool:
 
        words = str.split(" ")
        if not len(words) == len(pattern):
            return False
        
        mapping = dict()
        
        for i in range(len(words)):
            if pattern[i] not in mapping:
                if words[i] not in mapping.values(): 
                    mapping[pattern[i]] = words[i]
                else:
                    return False
            else:
                if not mapping[pattern[i]] == words[i]:
                    return False
        return True
                

File: 0290-word-pattern/0290-word-pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
    let map = new Map()
    let res=true
    s=s.split(" ")
    if(s.length !==pattern.length) return false
    if(new Set(s).size !== new Set(pattern).size) return false
    for(let i=0;i<pattern.length;i++){
       if(!map.has(pattern[i])){
           map.set(pattern[i],s[i])
       }
       
        if(s[i]!==map.get(pattern[i])) return false
    }    
    
   return res
    
};
